src/pytorch_lightning/core/module.py

```diff
@@ -1577,7 +1586,7 @@ def lr_scheduler_step(self, scheduler, optimizer_idx, metric):
 
         """
         if metric is None:
-            scheduler.step()
+            scheduler.step()  # type: ignore[call-arg]
```

Comment:
I think we should raise an error here when `metric=None` and the scheduler is `ReduceOnPlateau` instead of ignoring the type error. 
cc @carmocca

Reply:
previous convo with @rohitgr7 https://github.com/Lightning-AI/lightning/pull/13603#discussion_r918291207

Reply:
this is not possible since this is an internal call that is made based on the config returned from `configure_optimizers`.
and the user is expected to override this hook rather than calling it explictly.